src/utils/resolveUser.ts

```diff
@@ -1,40 +1,42 @@
-import { OAuth2Client } from "google-auth-library";
-import { TokenPayload } from "google-auth-library/build/src/auth/loginticket";
 import * as jwt from "jsonwebtoken";
-import { User, UserSessions } from "../generated/prisma-client";
-import { Context } from "prisma-client-lib/dist/types";
+import { Prisma, User, Admin } from "../generated/prisma-client";
 
-const client = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
-
-async function resolveUserUsingGoogle(ctx): Promise<User> {
-  try {
-    await client.verifyIdToken({
-      idToken: ctx.token,
-      audience: process.env.GOOGLE_CLIENT_ID,
-    });
-  } catch (e) {
-    return null;
-  }
-  const ticket = await client.verifyIdToken({
-    idToken: ctx.token,
-    audience: process.env.GOOGLE_CLIENT_ID,
-  });
-  const payload: TokenPayload = ticket.getPayload();
-  if (payload && payload.hd !== "connect.hku.hk") return null;
-  return ctx.prisma.user({ email: payload.email });
+interface SessionObject extends Object {
+  id: string;
+  createdAt?: string;
+  iat: number;
 }
 
-async function resolveUserUsingJWT(ctx: Context): Promise<User> | null {
+export const resolveUserUsingJWT = async (
+  prisma: Prisma,
+  token: string
+): Promise<any> => {
   try {
-    const userSession: UserSessions = jwt.verify(
-      ctx.token,
+    const sessionObject = jwt.verify(
+      token,
       process.env.PRISMA_SECRET
-    ) as UserSessions;
-    const sessionId = userSession.id;
-    return ctx.prisma.userSessions({ id: sessionId }).user();
+    ) as SessionObject;
+    const userSession = await prisma.userSessions({ id: sessionObject.id });
+    if (userSession === null) {
+      const adminSession = await prisma.adminUserSession({
+        id: sessionObject.id,
+      });
+      if (adminSession === null) {
+        return null;
+      } else {
+        const userObject = await prisma
+          .adminUserSession({ id: sessionObject.id })
+          .user();
+        return { user: userObject, priviledge: "admin" };
```

Comment:
its spelled `privilege` not `priviledge` right?

Reply:
oops